Extensions/retags.js

```diff
@@ -19,111 +19,168 @@ XKit.extensions.retags = {
 			this.blog_name = $('body').data('new-root').split('/').pop();
 		} catch(e) {}
 		this.add_toggle();
-		this.observer.observe($('body')[0],{childList:true,subtree:true});
+		this.observer.observe($('body')[0],{
+			childList:true,
+			subtree:true
+		});
 		this.tag(this.selectors);
 	},
 
 	observer: new MutationObserver(function(ms){
-		var inner, $el;
+		var $baseMutatedElements;
+		var notesProcessed = false;
 		ms.forEach(function(m){
-			$el = $(m.addedNodes);
-			inner = $el.find('.note:not(.ui_post_badge)');
-			if (inner.length > 0) {
-				XKit.extensions.retags.tag(inner.filter(XKit.extensions.retags.selectors));
+			$baseMutatedElements = $(m.addedNodes);
+			var nonPopoverNotes = $baseMutatedElements.find('.note:not(.ui_post_badge)');
+			if ($baseMutatedElements.hasClass('note-list')) {
+				//remove existing retags items (if tabbing between popover views)
+				$baseMutatedElements.find('.retags-wrapper').remove();
+				var elements = $baseMutatedElements.find('.post-activity-note');
+				notesProcessed = true;
+				XKit.extensions.retags.tag_popover(elements.filter(XKit.extensions.retags.selectors));
+			} else if ($baseMutatedElements.hasClass('post-activity-note') && !notesProcessed) {
```

Comment:
feels like the else if and the notesProcessed variable are redundant w/ each other.